site-scrapers/Hannaford.js

```diff
@@ -63,10 +65,77 @@ async function ScrapeWebsiteData(browser) {
                 const filename =
                     "hannaford-zip-" + loc.zip + "-date-" + today + ".png";
                 await page.screenshot({ path: filename });
+                await saveBodyHtml(page);
             }
             results[loc.zip] = result;
         }
     }
 
     return results;
 }
+
+async function writeFile(filename, data) {
+    return new Promise((resolve, reject) => {
+        fs.writeFile(filename, data, (err) => {
+            if (err) {
+                reject(err);
+            } else {
+                resolve();
+            }
+        });
+    });
+}
+
+async function saveBodyHtml(page) {
+    // It appears puppeteer does not a generic page.waitForNetworkIdle
+    // so for now we just assume all XHR requests will complete within
+    // a reasonable timeframe, for now 10s.
+    await new Promise((resolve) => {
+        setTimeout(function () {
+            resolve("done waiting");
+        }, 1000);
+    });
+
+    const body = await page.$("body");
+    const html = await body.evaluate((node) => node.outerHTML);
+    const timestamp =
+        new Date().toISOString().substring(0, 16).replace(":", "") + "Z";
+    const fileName = ["hannaford-", timestamp, ".html"].join("");
```

Comment:
nitpicky but a little more readable (imo): 
`hannaford-${timestamp}.html`
UGH can't figure out how to make the tick marks show. Tick marks at the beginning and end, like, in the code haha

Reply:
Also you need to add something unique - maybe the zip or something? - because right now if I unconditionally save for each location they all have the same timestamp (and therefore the same filename).
Verified it works though (the uploading)!

Reply:
late to the party, but fyi you can show template literals by putting the whole line in triple-ticks (```) (on a line before and after) to quote the singleticks:
```
const fileName = `hannaford-${timestamp}.html`;
```

Reply:
Good call!